[EA Forum only] add toggle to show/hide home page right-hand side

s-cheng opened the pull request:
This PR is branched off https://github.com/ForumMagnum/ForumMagnum/pull/7678. It adds a toggle on the right side of the screen that logged in users can use to hide or show the right-hand side column.

<img width="550" alt="Screen Shot 2023-08-07 at 8 15 25 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/10fa9dae-907e-453f-b70c-947de315ce8c">

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205229343800465) by [Unito](https://www.unito.io)

s-cheng commented on packages/lesswrong/components/Layout.tsx:
```diff
@@ -136,10 +136,10 @@ const styles = (theme: ThemeType): JssStyles => ({
         min-content
-        minmax(0, 1fr)
+        1fr
         min-content
         minmax(50px, max-content)
-        minmax(0, 1fr)
+        1fr
```
I think there's no reason to `minmax()` those two columns

Will-Howard reviewed (approved):
Nice! I like the little animation when you hover over it

agnesstenlund commented:
looks so good! 👏 thanks for really paying attention to all the details.

The only thing that felt surprising was the instant subscribe. Since it's just a button with no input I assumed there would be a second confirmation step or that I'd be asked for email. I even knew that it would in fact subscribe me instantly but I was still surprised 😅 the confirmation also doesn't say which email was subscribed, which probably added to the feeling of confusion/surprise.

I think we discussed this elsewhere but I think the ideal solution would be to auto-fill the input with the email we have (rather than just showing the button), to make it more clear to the user what's happening. Or just leave the field empty in case they'd like to subscribe with another email than the one they've signed up with

s-cheng commented:
@agnesstenlund Thanks! I've gone ahead and updated the digest ad a bit.

Logged in users that have an email address will now see it in the input (the input is disabled because it's just informational):
<img width="546" alt="Screen Shot 2023-08-11 at 3 27 48 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/e0b91189-4776-4b6d-8798-211c88e760a9">

And I added the narrower version of the button:
<img width="342" alt="Screen Shot 2023-08-11 at 3 28 09 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/e6adfa52-88a8-4a58-9283-33339aeb52c9">

After they submit, rather than showing the toast message, we update the ad to show a success message:
<img width="546" alt="Screen Shot 2023-08-11 at 3 28 26 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/cf37e73d-76e9-4635-a665-c9125643bf37">

The account settings link there goes to `/account?highlightField=subscribedToDigest` which highlights the relevant field on the page.

@Will-Howard FYI I also changed this to use localstorage instead of a cookie (I forgot that I used that in this case)

agnesstenlund commented:
I really like this! thanks so much 🙌 

Two questions:
1. About the success state: will the ad go away at some point? I like having the success message in context but would also want the banner to go away, at least at some point. May still be better to use the toast but include all of this information (I like that you included how to change the subscription)

2. Could we make the text editable when we auto-input? You say it shouldn't be since it's informational only, but I'd expect this text to be black and ideally editable, so you can change it if you want to subscribe with another email. If not editable I'd perhaps still make it black, since grey makes it look a lot like a placeholder and I'd probably be confused why I can't click into the input to add text

Another tiny thing I would follow-up with: 
* I'd remove the little Email adress* at the top–with a placeholder like `example@email.com` that feels really clear anyway (also since it's just one field it's less of a question of "which of these fields are mandatory")

s-cheng commented:
@agnesstenlund 💜 
1. The banner should be gone when they refresh or return to the page. I feel like the toast is not as good since it's pretty far away from the subscribe action, and I don't even always notice it happening. I think ideally the ad would collapse after some time, like 5 sec?
2. We try to track your digest subscription state both on the forum and on Mailchimp. Subscribing with an email that's different from your forum account's email address is not well supported (that's what I meant by "informational only"). In this case it's more complicated, because we actually show different forms depending on whether your account has an email address or not. It would take a bit of time to figure out how to properly handle this - we could look into it if we decide to release this to more users.
